File: src/Components/Logic.py
import random
from random import randrange 
from random import choice
from string import ascii_uppercase


def generate_key(length, digits = 1):
    #return(''.join( str(random.randint(0,1)) for i in range(length)))
    numbers = []
    for i in range(length):
        numbers.append(random.randint(0,(10**digits)-1))
    return(numbers)
                       
def generate_plaintext(length, digits = 1):
    numbers = []
    for i in range(length):
        numbers.append(random.randint(0,(10**digits)-1))
    return(numbers)

def perform_op(key, message, op, carry = 0):
  
    res = [] 
    for i in range(len(key)):
        
        if(op == "+"):
            print("eyyyyyyyyyyyyyyy")
            result = key[i] + message[i]
            print(result)
            if(carry):
                result = result % 10 
            res.append(result)
            
        elif(op == "-"):
            result = key[i] - message[i]
            if(carry):
                result = result % 10 
            res.append(result)
            
        elif(op == "*"):
            result = key[i] * message[i]
            if(carry):
                result = result % 10 
            res.append(result)
    return res 
  
plaintexts = (generate_plaintext(10), generate_plaintext(10))
b = random.randrange(0,2)

selected_text = plaintexts[b]
otp = generate_key(10)

print("key:" + str(otp))
print("b:" + str(b))
print("m0: " + str(plaintexts[0]))
print("m1: " + str(plaintexts[1]))
print("selected message" + str(selected_text))

print("\n\nmessage operation")
print(perform_op(otp, selected_text, "+", 1)) 

File: src/Components/Logic.js
// Different game modes
//
//       
//
// 1st game => n = 2, if digit greater than 9 then it wraps around (e.g 10 gives 0, 11 - 1)
// 
//
//
//
//
//




function startGame(){


}

function select_encrpytion_scheme(){
	// be able to selete the type ofeencryption

}



function switch_role(){
	// To be able switch teams 	
}


// Generates a key based on length and digits defined by the user
function generate_key(length, digits = 1){
    numbers = []
    var i;
    for(i = 0; i < length; i++){
    	numbers.push(Math.floor(Math.random()*Math.pow(10,digits)))
      console.log(numbers)
    }
  
    return(numbers)
}


// Geneartes plaintexts based on the users inputs for length of key and digits
function generate_plaintext(length, digits = 1){
    numbers = []
    var i;
    for(i = 0; i < length; i++){
    	numbers.push(Math.floor(Math.random()*Math.pow(10,digits)))
      console.log(numbers)
    }
  
    return(numbers)
}


// Perform encrpytion scheme based on given operation and carry flag 
function perform_op(key, message, op, carry = 0){
  
    res = []
    var i  
    for(i = 0; i < key.length; i++){
        
          if(op == "+"){
         
            result = key[i] + message[i]
            if(carry){
                result = result % 10 
            }
            res.push(result)
           }
	        else if(op == "-"){
	            result = key[i] - message[i]
            		if(carry){
                	result = result % 10 
                	}
            	res.push(result)
            }
        	else if(op == "*"){
            	result = key[i] * message[i]
            	if(carry) {
                	result = result % 10 
            	}
            	res.push(result)
        	}
        	
        }
     
    return res 
  }